tests/core/data/test_write_single_sheet_excel_data_node.py

```diff
@@ -0,0 +1,415 @@
+    excel_dn = ExcelDataNode(
+        "foo", Scope.SCENARIO, properties={"path": tmp_excel_file, "exposed_type": "numpy", "has_header": False}
+    )
+
+    arr = np.array([[1], [2], [3], [4], [5]])
+    excel_dn.write(arr)
+    excel_data = excel_dn.read()
+    assert isinstance(excel_data, dict) and "Sheet1" in excel_data.keys()
+    assert np.array_equal(excel_dn.read()["Sheet1"], arr)
+
+    arr = arr[0:3]
+    excel_dn.write(arr)
+    excel_data = excel_dn.read()
+    assert isinstance(excel_data, dict) and "Sheet1" in excel_data.keys()
+    assert np.array_equal(excel_dn.read()["Sheet1"], arr)
+
+    excel_dn.write(None)
+    excel_data = excel_dn.read()
+    assert isinstance(excel_data, dict) and "Sheet1" in excel_data.keys()
+    assert excel_dn.read()["Sheet1"].size == 0
+
+
+def test_write_with_header_single_sheet_custom_exposed_type_with_sheet_name(tmp_excel_file):
+    excel_dn = ExcelDataNode(
+        "foo",
+        Scope.SCENARIO,
+        properties={"path": tmp_excel_file, "sheet_name": "Sheet1", "exposed_type": MyCustomObject},
+    )
+
+    data = [MyCustomObject(0, 1, "hi"), MyCustomObject(1, 2, "world"), MyCustomObject(2, 3, "text")]
+    excel_dn.write(data)
+    assert all(actual == expected for actual, expected in zip(excel_dn.read(), data))
+
+    excel_dn.write(None)
+    assert excel_dn.read() == []
+
+
+def test_write_with_header_single_sheet_custom_exposed_type_without_sheet_name(tmp_excel_file):
+    excel_dn = ExcelDataNode("foo", Scope.SCENARIO, properties={"path": tmp_excel_file, "exposed_type": MyCustomObject})
+
+    data = [MyCustomObject(0, 1, "hi"), MyCustomObject(1, 2, "world"), MyCustomObject(2, 3, "text")]
+    excel_dn.write(data)
+    assert all(actual == expected for actual, expected in zip(excel_dn.read(), data))
+
+    excel_dn.write(None)
+    assert excel_dn.read() == []
+
+
+def test_write_without_header_single_sheet_custom_exposed_type_with_sheet_name(tmp_excel_file):
+    excel_dn = ExcelDataNode(
+        "foo",
+        Scope.SCENARIO,
+        properties={
+            "path": tmp_excel_file,
+            "sheet_name": "Sheet1",
+            "exposed_type": MyCustomObject,
+            "has_header": False,
+        },
+    )
+
+    data = [MyCustomObject(0, 1, "hi"), MyCustomObject(1, 2, "world"), MyCustomObject(2, 3, "text")]
+    excel_dn.write(data)
+    assert all(actual == expected for actual, expected in zip(excel_dn.read(), data))
+
+    excel_dn.write(None)
+    assert excel_dn.read() == []
+
+
+def test_write_without_header_single_sheet_custom_exposed_type_without_sheet_name(tmp_excel_file):
+    excel_dn = ExcelDataNode(
+        "foo", Scope.SCENARIO, properties={"path": tmp_excel_file, "exposed_type": MyCustomObject, "has_header": False}
+    )
+
+    data = [MyCustomObject(0, 1, "hi"), MyCustomObject(1, 2, "world"), MyCustomObject(2, 3, "text")]
+    excel_dn.write(data)
+    assert all(actual == expected for actual, expected in zip(excel_dn.read(), data))
```

Comment:
this behavior is inconsistent with what we have for numpy (`test_write_without_header_single_sheet_numpy_without_sheet_name()`) and pandas (test_write_without_header_single_sheet_pandas_without_sheet_name) as it's returning a list of value directly  instead of a dictionary. We should discuss this within a separate ticket.

Reply:
https://github.com/orgs/Avaiga/projects/6/views/2?pane=issue&itemId=52334690